src/main/java/com/google/api/generator/engine/ast/JavaDocComment.java

```diff
@@ -0,0 +1,105 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.api.generator.engine.ast;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@AutoValue
+public abstract class JavaDocComment implements Comment {
+
+  public abstract ImmutableList<String> comments();
+
+  public static Builder builder() {
+    return new AutoValue_JavaDocComment.Builder();
+  }
+
+  public static JavaDocComment withComment(String comment) {
+    return JavaDocComment.builder().addComment(comment).build();
+  }
+
+  @AutoValue.Builder
+  public abstract static class Builder {
+    protected abstract ImmutableList.Builder<String> commentsBuilder();
+
+    public Builder setDeprecated(String deprecatedText) {
+      commentsBuilder().add(String.format("%s %s", "@deprecated", deprecatedText));
+      return this;
+    }
+
+    public Builder addComment(String comment) {
+      commentsBuilder().add(comment);
+      return this;
+    }
+
+    public Builder addParam(String name, String description) {
+      String parameter = String.format("%s %s %s", "@param", name, description);
+      commentsBuilder().add(parameter);
+      return this;
+    }
+
+    public Builder setThrows(String type, String description) {
+      String throwsText = String.format("%s %s %s", "@throws", type, description);
+      commentsBuilder().add(throwsText);
+      return this;
+    }
+
+    public Builder addSampleCode(String sampleCode) {
+      commentsBuilder().add("<pre><code>");
+      String[] sampleLines = sampleCode.split("\\r?\\n");
```

Comment:
Could you please see if this (and the for-loops below) can be replaced with stream() equivalents?

Reply:
done.

Reply:
There are tradeoffs to streams versus for-loops (see a discussion [here](https://stackoverflow.com/questions/44180101/in-java-what-are-the-advantages-of-streams-over-loops)), and IMO the benefits of streams make them a better choice.

Reply:
Thank you for the clarification!